service/lib/agama/autoyast/software_reader.rb

```diff
@@ -37,12 +37,16 @@ def initialize(profile)
       #
       # @return [Hash] Agama "software" section
       def read
-        return {} if profile["software"].nil?
-
         software = {}
 
-        software["patterns"] = profile["software"].fetch_as_array("patterns")
-        software["packages"] = profile["software"].fetch_as_array("packages")
+        if profile["software"]
+          software["patterns"] = profile["software"].fetch_as_array("patterns")
+          software["packages"] = profile["software"].fetch_as_array("packages")
+        end
+        if profile["add-on"]
+          repos = process_repos
```

Comment:
NP: I do not know if that logic should live here. A bit disconnected of that part currently. But I'd at least rename it to `fecth_addon_repos` to make it more readable.